src/parquet/column/properties.h

```diff
@@ -151,76 +215,101 @@ class PARQUET_EXPORT WriterProperties {
+              dictionary_enabled_, dictionary_pagesize_, pagesize_, version_, created_by_,
+              default_encoding_, encodings_, default_codec_, codecs_));
     }
 
    private:
     MemoryAllocator* allocator_;
+    bool dictionary_enabled_default_;
+    std::unordered_map<std::string, bool> dictionary_enabled_;
     int64_t dictionary_pagesize_;
+    int64_t pagesize_;
+    ParquetVersion::type version_;
+    std::string created_by_;
     // Encoding used for each column if not a specialized one is defined as
     // part of encodings_
     Encoding::type default_encoding_;
     std::unordered_map<std::string, Encoding::type> encodings_;
-    int64_t pagesize_;
-    ParquetVersion::type version_;
     // Default compression codec. This will be used for all columns that do
     // not have a specific codec set as part of codecs_
     Compression::type default_codec_;
     ColumnCodecs codecs_;
   };
 
-  MemoryAllocator* allocator() const { return allocator_; }
+  inline MemoryAllocator* allocator() const { return allocator_; }
 
-  int64_t dictionary_pagesize() const { return dictionary_pagesize_; }
+  inline bool dictionary_enabled(const std::shared_ptr<schema::ColumnPath>& path) const {
+    auto it = dictionary_enabled_.find(path->ToDotString());
+    if (it != dictionary_enabled_.end()) { return it->second; }
+    return dictionary_enabled_default_;
+  }
+
+  inline int64_t dictionary_pagesize() const { return dictionary_pagesize_; }
 
-  int64_t data_pagesize() const { return pagesize_; }
+  inline int64_t data_pagesize() const { return pagesize_; }
 
-  ParquetVersion::type version() const { return parquet_version_; }
+  inline ParquetVersion::type version() const { return parquet_version_; }
 
-  Encoding::type encoding(const std::shared_ptr<schema::ColumnPath>& path) const {
-    Encoding::type coding = default_encoding_;
+  inline std::string created_by() const { return parquet_created_by_; }
+
+  inline Encoding::type encoding(const std::shared_ptr<schema::ColumnPath>& path) const {
     auto it = encodings_.find(path->ToDotString());
-    if (it != encodings_.end()) { coding = it->second; }
-
-    // Use the correct enum value for dictionary coding based on the used Parquet version
-    if (coding == Encoding::PLAIN_DICTIONARY || coding == Encoding::RLE_DICTIONARY) {
-      if (parquet_version_ == ParquetVersion::PARQUET_1_0) {
-        return Encoding::PLAIN_DICTIONARY;
-      } else {
-        return Encoding::RLE_DICTIONARY;
-      }
+    if (it != encodings_.end()) { return it->second; }
+    return default_encoding_;
+  }
+
+  inline Encoding::type dictionary_encoding() const {
+    if (parquet_version_ == ParquetVersion::PARQUET_1_0) {
+      return Encoding::PLAIN_DICTIONARY;
+    } else {
+      return Encoding::RLE_DICTIONARY;
     }
-    return coding;
   }
 
-  Compression::type compression(const std::shared_ptr<schema::ColumnPath>& path) const {
+  inline Encoding::type dictionary_index_encoding() const {
```

Comment:
Are these names flipped? the `dictionary_encoding` method above returns the encoding set on the data pages, also referred to as the "indices". Maybe call this method "dictionary_page_encoding" and the other "dictionary_index_encoding"?